Source/PeleLMEos.cpp

```diff
@@ -298,19 +320,29 @@ PeleLM::adjustPandDivU(std::unique_ptr<AdvanceAdvData> &advData)
 
     // Adjust
     for (int lev = 0; lev <= finest_level; ++lev) {
+        // ThetaHalft is now delta_theta
         ThetaHalft[lev]->plus(-Thetabar,0,1);
+        // mac_divu is now delta_S
         advData->mac_divu[lev].plus(-Sbar,0,1);
     }
+  
+    // Compute 1/Volume * int(U_inflow)dA across all boundary faces
+    amrex::Real umacFluxBalance = AMREX_D_TERM(  m_domainUmacFlux[0] + m_domainUmacFlux[1],
+                                       + m_domainUmacFlux[2] + m_domainUmacFlux[3],               
+                                       + m_domainUmacFlux[4] + m_domainUmacFlux[5]);
+    Real divu_vol = umacFluxBalance/m_uncoveredVol;
 
     // Advance the ambient pressure
-    m_pNew = m_pOld + m_dt * (Sbar/Thetabar);
-    m_dp0dt = Sbar/Thetabar;
+    m_pNew = m_pOld + m_dt * (Sbar - divu_vol)/Thetabar;
+    m_dp0dt = (Sbar - divu_vol)/Thetabar;
+
 
     // subtract \tilde{theta} * Sbar / Thetabar from divu
     for (int lev = 0; lev <= finest_level; ++lev) {
 #ifdef AMREX_USE_OMP
 #pragma omp parallel if (Gpu::notInLaunchRegion())
 #endif
+
```

Comment:
This empty line might mess with #pragma omp parallel if

Reply:
deleting...